### PR TITLE
Include encoded forms in EventListener APIs

### PR DESCRIPTION
**Pull request by swankjesse**
I'm hoping this is sufficient to build a very powerful
DevelopmentEventListener that can trace leaks precisely.

### CODE REVIEW

**Comment by swankjesse** [comment]
This needs tests!

File: zipline/src/commonMain/kotlin/app/cash/zipline/internal/bridge/InboundService.kt [hunk 1]
```diff
@@ -32,57 +33,65 @@ internal class InboundService<T : ZiplineService>(
...
-    return endpoint.json.encodeToStringFast(
-      (call.function?.kotlinResultSerializer ?: failureResultSerializer) as KSerializer<Any?>,
-      result,
-    )
+    val theResult = try {
+      val success = function.call(service, internalCall.args)
+      Result.success(success)
+    } catch (e: Throwable) {
+      Result.failure(e)
+    }
```
Comment (JakeWharton) [inline zipline/src/commonMain/kotlin/app/cash/zipline/internal/bridge/InboundService.kt]: This pattern is `runCatching { .. }`

File: zipline/src/commonMain/kotlin/app/cash/zipline/internal/bridge/InboundService.kt [hunk 2]
```diff
@@ -32,57 +33,65 @@ internal class InboundService<T : ZiplineService>(
...
+      if (function == null) {
+        result = Result.failure(unexpectedFunction(internalCall.functionName))
+      } else {
+        val callStart = endpoint.eventListener.callStart(externalCall)
+        result = try {
+          val success = function.callSuspending(service, args)
+          Result.success(success)
+        } catch (e: Throwable) {
+          Result.failure(e)
         }
```
Comment (JakeWharton) [inline zipline/src/commonMain/kotlin/app/cash/zipline/internal/bridge/InboundService.kt]: Same pattern

**Review by JakeWharton — APPROVED** [review]

File: zipline/src/commonMain/kotlin/app/cash/zipline/internal/bridge/InboundService.kt (hunk 1 above)
Comment (swankjesse) [reply zipline/src/commonMain/kotlin/app/cash/zipline/internal/bridge/InboundService.kt]: Ooooh yeah I like that.

File: zipline/src/commonMain/kotlin/app/cash/zipline/internal/bridge/InboundService.kt (hunk 2 above)
Comment (swankjesse) [reply zipline/src/commonMain/kotlin/app/cash/zipline/internal/bridge/InboundService.kt]: Done

**Pull request merged**